rfcs/rfc-using-pinia-for-form-components.md

```diff
@@ -0,0 +1,318 @@
+   * */
+  setProp (store, nestedProperty, value) {
+    const prop = nestedProperty.shift()
+    if (!store[prop]) {
+      Vue.set(store, prop, {})
+    }
+    if (!nestedProperty.length) {
+      if (value && typeof value === 'object' && !Array.isArray(value)) {
+        store[prop] = { ...store[prop], ...value }
+      } else {
+        store[prop] = value
+      }
+      return
+    }
+    this.setProp(store[prop], nestedProperty, value)
+  }
+```
+
+We envision expanding the low-level components like `SingleCheckbox.vue` to include component
+to display radio buttons, text fields and date fields etc.  Each of these components would talk
+directly to the store using the store getter `getNestedAttribute` and the mutator `setNestedAttribute`.
+
+The new approach has more advantages as the forms get more complex and have more nested objects.  For
+example:
+
+```html
+<div>
+  <form-row>
+    <in-line-check-box id="corp_owner" :path="path" :option="true">
+        Owned by corporate entity
+    </in-line-check-box>
+  </form-row>
+  <!-- corporation sub-component start -->
+  <form-row v-if="getNestedAttribute(path, 'corp_owner') === true">
+    <text-field id="name" :path="path + '.corporation'" rules="max:40">
+      Corporation Name</text-field>
+  </form-row>
+  <!-- corporation sub-component start -->
+  <!-- individual sub-component start -->
+  <form-row v-if="getNestedAttribute(path, 'corp_owner') !== true">
+    <text-field id="last_name" :path="path + '.individual'" rules="max:20">
+      Owner's Last Name</text-field>
+    <text-field id="first_name" :path="path + '.individual'" rules="max:20">
+      Owner's First Name</text-field>
+  </form-row>
+  <!-- individual sub-component end -->
+  <form-row>
+    <text-field id="address" :path="path" placeholder="Address" rules="max:25">
+        Address Line
+    </text-field>
+  </form-row>
+  <form-row>
+    <text-field id="city" :path=path rules="max:20">City</text-field>
+    <province-field id="province" :path=path>Prov / State</province-field>
+    <text-field id="postal" :path=path>Postal / Zip</text-field>
+    <phone-field id="phone" :path=path rules="phone">Phone</phone-field>
+  </form-row>
+</div>
+```
+
+
+# Drawbacks
+
+We don't see this proposal as having significant drawbacks. If adopted, this new
+approach could be implemented gradually and could be used alongside the current approach.
+
+Some developers may have concerns about moving the business logic from components to the store,
+but we suggest using Pinia for business logic is better and more flexible than storing 
+business logic in components. We'd be interested in what the group's feedback might be on this topic.
+
+Form validation is a big part of developing reusable form components. We haven't mentioned form
+validation in this RFC, but would like to do so in a future RFC.
+
+# Alternatives
+
+We could use the Composition API instead of Pinia. The Composition API is native to Vue and has
```

Comment:
Why wouldn't you use Composition API even if you're using Pinia?

Reply:
That's a good point @seeker25 There's nothing preventing us from using the Composition API alongside Pinia/Vuex.